Emit CC problems for unsupported listeners in BuildEventsListenerRegistry (#33747)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

Previously, these tasks were automatically added to the project for every build.
These tasks are now only added to a project when a rule-based plugin is applied (such as those provided by Gradle's support for building native software).

> === Configuration Cache changes

Reviewer: ```suggestion
=== Configuration Cache changes

```

> ==== Unsupported build events listeners are now configuration cache problems

Reviewer: ```suggestion
==== Unsupported build event listeners are now configuration cache problems
```

The build event listener registration method
link:{javadocPath}/org/gradle/build/event/BuildEventsListenerRegistry.html#onTaskCompletion(org.gradle.api.provider.Provider)[`BuildEventsListenerRegistry.onTaskCompletion`]
accepts arbitrary providers of any
link:{javadocPath}/org/gradle/tooling/events/OperationCompletionListener.html[`OperationCompletionListener`]
implementations.
However, only providers returned from
link:{javadocPath}/org/gradle/api/services/BuildServiceRegistry.html#registerIfAbsent(java.lang.String,java.lang.Class,org.gradle.api.Action)[`BuildServiceRegistry.registerIfAbsent`]
or
link:{javadocPath}/org/gradle/api/services/BuildServiceRegistration.html#getService()[`BuildServiceRegistration.getService`]
are currently supported when the Configuration Cache is enabled.

Previously, unsupported providers were silently discarded and never received events when the configuration cache was used.

> Starting with Gradle 9.0, registering such providers becomes a configuration cache problem and fails the build.

Reviewer: ```suggestion
Starting with Gradle 9, registering such providers now causes a configuration cache problem and fails the build.
```

> If your build was correctly with configuration cache before (for example, the registered listeners weren't crucial for build correctness),

Reviewer: ```suggestion
If your build was previously working with the configuration cache (e.g., the listeners were nonessential),
```

> you can temporarily revert to old behavior of ignoring such providers.

Reviewer: ```suggestion
you can temporarily revert to the old behavior by setting:
```

> To do that, set the Gradle property `org.gradle.configuration-cache.unsafe.ignore.unsupported-build-events-listeners` to `true`.

Reviewer: ```suggestion
[source,properties]
----
org.gradle.configuration-cache.unsafe.ignore.unsupported-build-events-listeners=true
----
```
